### ISSUE TITLE
NUMERIC(11,2) translated to bigint(19), rounding the values to integers

### ISSUE THREAD

**Issue by guidot** [techouse/sqlite3-to-mysql#125]
**Describe the bug**
A NUMERIC(11,2) field in an sqlite table got translated to bigint(19) in MariaDB. As a result, the floating point values contained therein were rounded to integers.

```sql
sqlite> PRAGMA table_info(invoices);
0|InvoiceId|INTEGER|1||1
...
8|Total|NUMERIC(10,2)|1||0

MariaDB> SHOW columns FROM invoices;
+-------------------+-------------+------+-----+---------+----------------+
| Field             | Type        | Null | Key | Default | Extra          |
+-------------------+-------------+------+-----+---------+----------------+
| InvoiceId         | int(11)     | NO   | PRI | NULL    | auto_increment |
...
| Total             | bigint(19)  | NO   |     | NULL    |                |
+-------------------+-------------+------+-----+---------+----------------+
```

```sql
sqlite> SELECT DISTINCT typeof(Total) FROM invoices;
real
```

```sql
sqlite> SELECT Total FROM invoices ORDER BY InvoiceId LIMIT 5;
1.98
3.96
5.94
8.91
13.86

MariaDB> SELECT Total FROM invoices ORDER BY InvoiceId LIMIT 5;
+-------+
| Total |
+-------+
|     2 |
|     4 |
|     6 |
|     9 |
|    14 |
+-------+

```
It looks like SQLite's [dynamic type system](https://sqlite.org/datatype3.html) has not been taken into account:
```python
if data_type in {"INT64", "NUMERIC"}:
    return "BIGINT" + self._column_type_length(column_type, 19)
```
**Expected behaviour**
The field created in MariaDB is of an adaquate numeric type that stores the data unchanged.

**Actual result**
The field created in MariaDB is of type bigint(19), which causes the values to be rounded to integers.

**System Information**

```bash
$ sqlite3mysql --version
```

```
| software               | version                                                                           |
|------------------------|-----------------------------------------------------------------------------------|
| sqlite3-to-mysql       | 2.1.7                                                                             |
|                        |                                                                                   |
| Operating System       | Linux 6.10.5-1-default                                                            |
| Python                 | CPython 3.11.9                                                                    |
| MySQL                  | mysql from 11.4.2-MariaDB, client 15.2 for Linux (x86_64) using  EditLine wrapper |
| SQLite                 | 3.46.0                                                                            |
|                        |                                                                                   |
| click                  | 8.1.7                                                                             |
| mysql-connector-python | 8.2.0                                                                             |
| pytimeparse2           | 1.7.1                                                                             |
| simplejson             | 3.19.2                                                                            |
| tabulate               | 0.9.0                                                                             |
| tqdm                   | 4.66.4                                                                            |
```

**Comment by guidot** [comment]
I don't know. I didn't create the db. Why does it matter? It's a valid type, isn't it?

**Comment by techouse** [comment]
Valid, yes, correct, debatable. SQLite is very lenient when it comes to this.

If you want it to work as is you'll have to change the original data type to `DECIMAL(10,2)` or wait until I add support for yet another non-standard conversion.

**Comment by techouse** [comment]
@guidot fix will be released in v2.3.1 shortly

**Comment by guidot** [comment]
That was quick!  Thanks a lot. I'll check it out.
